langserver/diagnostics.go

```diff
@@ -18,18 +19,37 @@ import (
 
 type diagnostics map[string][]*lsp.Diagnostic // map of URI to diagnostics (for PublishDiagnosticParams)
 
+type diagnosticsCache struct {
+	cache diagnostics
+	lock  sync.Mutex
```

Comment:
nit: use style of a "mutex hat", so call rename `lock` to `mu` and have it above cache. https://dmitri.shuralyov.com/idiomatic-go#mutex-hat